pre_cleanup.lua

```diff
@@ -0,0 +1,47 @@
+local os = require('os')
+
+local function clean_state(self)
+    box.space._space:pairs():map(function(tuple)
+        return tuple[3]
+    end):filter(function(name)
+        local first_char = string.sub(name, 1, 1)
+        return first_char ~= '_'
+    end):each(function(name)
+        box.space[name]:drop()
+    end)
+    local allowed_users = {
+        guest = true,
+        admin = true,
+    }
+
+    box.space._user:pairs():filter(function(tuple)
+        return tuple[4] == 'user'
+    end):map(function(tuple)
+        return tuple[3]
+    end):filter(function(name)
+        return not allowed_users[name]
+    end):each(function(name)
+        box.schema.user.drop(name)
+    end)
+    local allowed_roles = {
+        public = true,
+        replication = true,
+        super = true,
+    }
+
+    box.space._user:pairs():filter(function(tuple)
+        return tuple[4] == 'role'
+    end):map(function(tuple)
+        return tuple[3]
+    end):filter(function(name)
+        return not allowed_roles[name]
+    end):each(function(name)
+        box.schema.role.drop(name)
+    end)
+
+    os.execute(string.format('rm -rf *.snap *.xlog'))
+end
+
```

Comment:
It should be good to have asserts like so: https://gist.github.com/bigbes/47e9e5c5cfb9245dbda94fd0d6c3ed11#file-example-lua-L46 to check that the clean up really do its work.